File: 백준/Gold/2473. 세 용액/세 용액.java
import java.io.*;
import java.util.*;

public class Main {
	static int n;
	static long[] arr;
	static int answer;

	public static void main(String[] args) throws Exception {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		StringTokenizer st;

		n = Integer.parseInt(bf.readLine());

		arr = new long[n];

		st = new StringTokenizer(bf.readLine());
		for (int i = 0; i < n; i++) {
			arr[i] = Long.parseLong(st.nextToken());
		}

		Arrays.sort(arr);

		long[] res = new long[3];

		long min = 4000000001L;

		for (int i = 0; i < n - 2; i++) {
			
			int left = i+1;
			int right = n - 1;
			
			while (left < right) {
				long temp = arr[left] + arr[right] + arr[i];
				long value = Math.abs(temp);

				if (min > value) {
					min = value;
					res[0] = arr[left];
					res[1] = arr[right];
					res[2] = arr[i];
				}

				if (temp > 0) {
					right--;
				} else {
					left++;
				}
			}
		}
		Arrays.sort(res);

		for(int i=0; i<3; i++) {
			sb.append(res[i]).append(" ");
		}
		System.out.println(sb.toString());
	}

}

File: 백준/Gold/2473. 세 용액/세 용액.js
const path = process.platform === 'linux' ? '/dev/stdin' : 'index.txt';
const input = require('fs').readFileSync(path).toString().trim().split('\n');

const n = +input[0];
const arr = input[1].split(' ').map(Number);

arr.sort((a, b) => a - b);

let answer = [];
let min = Infinity;

for (let i = 0; i < n; i++) {
  const target = arr[i];

  const newArr = [...arr.slice(0, i), ...arr.slice(i + 1)];
  let left = 0;
  let right = n - 2;

  while (left < right) {
    const sum = newArr[left] + newArr[right] + target;

    if (Math.abs(sum) <= min) {
      answer = [target, newArr[left], newArr[right]];
      min = Math.abs(sum);
    }

    if (sum >= 0) {
      right--;
    } else {
      left++;
    }
  }
}

console.log(answer.sort((a, b) => a - b).join(' '));

/**
 *
 * 5
 *
 * -2 6 -97 -6 98
 *
 *
 */